packages/flutter/lib/src/widgets/scroll_metrics.dart

```diff
@@ -64,21 +62,21 @@ abstract class ScrollMetrics {
   ///
   /// This value should typically be non-null and less than or equal to
   /// [maxScrollExtent]. It can be negative infinity, if the scroll is unbounded.
-  double get minScrollExtent;
+  double? get minScrollExtent;
```

Comment:
how much trouble do we get in if all these are non-nullable?

Reply:
Trying to make it non-nullable I face test failures:

```
When the exception was thrown, this was the stack:
#2      new _CastError._create (dart:core-patch/errors_patch.dart:110:27)
#3      ScrollPosition.minScrollExtent (package:flutter/src/widgets/scroll_position.dart:139:49)
#4      ScrollMetrics.copyWith (package:flutter/src/widgets/scroll_metrics.dart:51:48)
#5      ScrollPosition.recommendDeferredLoading (package:flutter/src/widgets/scroll_position.dart:861:7)
#6      Scrollable.recommendDeferredLoadingForContext (package:flutter/src/widgets/scrollable.dart:296:28)
#7      ScrollAwareImageProvider.resolveStreamForKey (package:flutter/src/widgets/scroll_aware_image_provider.dart:98:20)
```

Reply:
looks like it's the same thing about delayed setup of the initial value, though this one seems harder to deal with so let's just let it be nullable for now i guess.